exporter/trace/stackdriver/stackdriver.go

```diff
@@ -0,0 +1,258 @@
+// relies on.
+func WithContext(ctx context.Context) func(o *options) {
+	return func(o *options) {
+		o.Context = ctx
+	}
+}
+
+func (o *options) handleError(err error) {
+	if o.OnError != nil {
+		o.OnError(err)
+		return
+	}
+	log.Printf("Failed to export to Stackdriver: %v", err)
+}
+
+const defaultTimeout = 5 * time.Second
+
+// Exporter is a trace exporter that uploads data to Stackdriver.
+//
+// TODO(yoshifumi): add a metrics exporter once the spec definition
+// process and the sampler implementation are done.
+type Exporter struct {
+	traceExporter *traceExporter
+}
+
+// NewExporter creates a new Exporter thats implements trace.Exporter.
+//
+// TODO(yoshifumi): add a metrics exporter one the spec definition
+// process and the sampler implementation are done.
+func NewExporter(opts ...Option) (*Exporter, error) {
+	o := options{}
+	for _, opt := range opts {
+		opt(&o)
+	}
+	if o.ProjectID == "" {
+		ctx := o.Context
+		if ctx == nil {
+			ctx = context.Background()
+		}
+		creds, err := google.FindDefaultCredentials(ctx, traceapi.DefaultAuthScopes()...)
+		if err != nil {
+			return nil, fmt.Errorf("Stackdriver: %v", err)
+		}
+		if creds.ProjectID == "" {
+			return nil, errors.New("Stackdriver: no project found with application default credentials")
+		}
+		o.ProjectID = creds.ProjectID
+	}
+	te, err := newTraceExporter(&o)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Exporter{
+		traceExporter: te,
+	}, nil
+}
+
+func newContextWithTimeout(ctx context.Context, timeout time.Duration) (context.Context, func()) {
+	if ctx == nil {
+		ctx = context.Background()
+	}
+	if timeout <= 0 {
+		timeout = defaultTimeout
+	}
+	return context.WithTimeout(ctx, timeout)
+}
+
+// Register registers e to the tracer.
+func (e *Exporter) Register() {
+	trace.Register()
+	trace.RegisterExporter(e)
+}
+
+// ExportSpan exports a SpanData to Stackdriver Trace.
+func (e *Exporter) ExportSpan(sd *trace.SpanData) {
+	if len(e.traceExporter.o.DefaultTraceAttributes) > 0 {
+		sd = e.sdWithDefaultTraceAttributes(sd)
+	}
+	e.traceExporter.ExportSpan(sd)
+}
+
+func (e *Exporter) sdWithDefaultTraceAttributes(sd *trace.SpanData) *trace.SpanData {
+	newSD := *sd
+	for k, v := range e.traceExporter.o.DefaultTraceAttributes {
+		switch val := v.(type) {
+		case bool:
+			newSD.Attributes = append(newSD.Attributes, key.New(k).Bool(val))
+		case int64:
+			newSD.Attributes = append(newSD.Attributes, key.New(k).Int64(val))
+		case float64:
+			newSD.Attributes = append(newSD.Attributes, key.New(k).Float64(val))
+		case string:
+			newSD.Attributes = append(newSD.Attributes, key.New(k).String(val))
+		}
+	}
+	newSD.Attributes = append(newSD.Attributes, sd.Attributes...)
+	return &newSD
+}
+
+// Flush waits for exported data to be uploaded.
+//
+// This is useful if your program is ending and you do not
+// want to lose recent stats or spans.
+func (e *Exporter) Flush() {
+	e.traceExporter.Flush()
+}
```

Comment:
If BatchSpanProcessor (bsp) is used then Flush() is not required. Shutdown() call in bsp is equivalent.

Reply:
ACK. Changed the method name from Flush to Shutdown.
https://godoc.org/go.opentelemetry.io/sdk/trace#BatchSpanProcessor.Shutdown
https://godoc.org/go.opentelemetry.io/sdk/trace#SimpleSpanProcessor.Shutdown